### ISSUE TITLE
Problems animating objects with `.add_background_rectangle()`

### ISSUE THREAD

**Issue by Elteoremadebeethoven** [ManimCommunity/manim#2337]
## Description of bug / unexpected behavior

If you add a background to a Mobject using `.add_background_rectangle()` there are some animations, like `Write`, that generate an error (log below)

## Expected behavior

This works (using `Create`):

```python
class Test1(Scene):
    def construct(self):
        t = Text("Some text")
        t.add_background_rectangle(color=TEAL)

        self.play(Create(t))
        self.wait()
```

https://user-images.githubusercontent.com/43224662/143748095-aeaed452-994a-428b-825d-71f93b929e7e.mp4

## How to reproduce the issue

But if you change `Create` with `Write`:

<details><summary>Code for reproducing the problem</summary>

```py
class Test1(Scene):
    def construct(self):
        t = Text("Some text")
        t.add_background_rectangle(color=TEAL)

        self.play(Write(t))
        self.wait()
```

</details>

It generates the error: 

```log
TypeError: set_style() got an unexpected keyword argument 'stroke_opacity'
```

## Logs
<details><summary>Full log</summary>

```
Manim Community v0.12.0

[11/28/21 03:11:40] INFO     Log file will be saved in media/logs/after_Test1.log                                                                          logger_utils.py:182
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/cli/render/commands.py:139 in render   │
│                                                                                                  │
│   136 │   │   for SceneClass in scene_classes_from_file(file):                                   │
│   137 │   │   │   try:                                                                           │
│   138 │   │   │   │   scene = SceneClass()                                                       │
│ ❱ 139 │   │   │   │   scene.render()                                                             │
│   140 │   │   │   except Exception:                                                              │
│   141 │   │   │   │   error_console.print_exception()                                            │
│   142 │   │   │   │   sys.exit(1)                                                                │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/scene/scene.py:216 in render           │
│                                                                                                  │
│    213 │   │   """                                                                               │
│    214 │   │   self.setup()                                                                      │
│    215 │   │   try:                                                                              │
│ ❱  216 │   │   │   self.construct()                                                              │
│    217 │   │   except EndSceneEarlyException:                                                    │
│    218 │   │   │   pass                                                                          │
│    219 │   │   except RerunSceneException as e:                                                  │
│                                                                                                  │
│ /home/alex/ManimCE/after.py:177 in construct                                                     │
│                                                                                                  │
│   174 │   │   t = Text("Some text")                                                              │
│   175 │   │   t.add_background_rectangle(color=TEAL)                                             │
│   176 │   │                                                                                      │
│ ❱ 177 │   │   self.play(Write(t))                                                                │
│   178 │   │   self.wait()                                                                        │
│   179                                                                                            │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/scene/scene.py:903 in play             │
│                                                                                                  │
│    900 │   │   │   return np.max([animation.run_time for animation in animations])               │
│    901 │                                                                                         │
│    902 │   def play(self, *args, **kwargs):                                                      │
│ ❱  903 │   │   self.renderer.play(self, *args, **kwargs)                                         │
│    904 │                                                                                         │
│    905 │   def wait(self, duration=DEFAULT_WAIT_TIME, stop_condition=None):                      │
│    906 │   │   self.play(Wait(run_time=duration, stop_condition=stop_condition))                 │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/renderer/cairo_renderer.py:92 in play  │
│                                                                                                  │
│    89 │   │   self.static_image = self.save_static_frame_data(scene, scene.static_mobjects)      │
│    90 │   │                                                                                      │
│    91 │   │   self.file_writer.begin_animation(not self.skip_animations)                         │
│ ❱  92 │   │   scene.begin_animations()                                                           │
│    93 │   │   if scene.is_current_animation_frozen_frame():                                      │
│    94 │   │   │   self.update_frame(scene)                                                       │
│    95 │   │   │   # self.duration stands for the total run time of all the animations.           │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/scene/scene.py:976 in begin_animations │
│                                                                                                  │
│    973 │   def begin_animations(self) -> None:                                                   │
│    974 │   │   """Start the animations of the scene."""                                          │
│    975 │   │   for animation in self.animations:                                                 │
│ ❱  976 │   │   │   animation.begin()                                                             │
│    977 │                                                                                         │
│    978 │   def is_current_animation_frozen_frame(self) -> bool:                                  │
│    979 │   │   """Returns whether the current animation produces a static frame (generally a Wa  │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/animation/creation.py:339 in begin     │
│                                                                                                  │
│   336 │   def begin(self) -> None:                                                               │
│   337 │   │   if self.reverse:                                                                   │
│   338 │   │   │   self.reverse_submobjects()                                                     │
│ ❱ 339 │   │   super().begin()                                                                    │
│   340 │                                                                                          │
│   341 │   def finish(self) -> None:                                                              │
│   342 │   │   super().finish()                                                                   │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/animation/creation.py:240 in begin     │
│                                                                                                  │
│   237 │                                                                                          │
│   238 │   def begin(self) -> None:                                                               │
│   239 │   │   self.outline = self.get_outline()                                                  │
│ ❱ 240 │   │   super().begin()                                                                    │
│   241 │                                                                                          │
│   242 │   def get_outline(self) -> Mobject:                                                      │
│   243 │   │   outline = self.mobject.copy()                                                      │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/animation/animation.py:194 in begin    │
│                                                                                                  │
│   191 │   │   │   # the internal updaters of self.starting_mobject,                              │
│   192 │   │   │   # or any others among self.get_all_mobjects()                                  │
│   193 │   │   │   self.mobject.suspend_updating()                                                │
│ ❱ 194 │   │   self.interpolate(0)                                                                │
│   195 │                                                                                          │
│   196 │   def finish(self) -> None:                                                              │
│   197 │   │   # TODO: begin and finish should require a scene as parameter.                      │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/animation/animation.py:294 in          │
│ interpolate                                                                                      │
│                                                                                                  │
│   291 │   │   │   The relative time to set the animation to, 0 meaning the start, 1 meaning      │
│   292 │   │   │   the end.                                                                       │
│   293 │   │   """                                                                                │
│ ❱ 294 │   │   self.interpolate_mobject(alpha)                                                    │
│   295 │                                                                                          │
│   296 │   def interpolate_mobject(self, alpha: float) -> None:                                   │
│   297 │   │   """Interpolates the mobject of the :class:`Animation` based on alpha value.        │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/animation/animation.py:309 in          │
│ interpolate_mobject                                                                              │
│                                                                                                  │
│   306 │   │   families = list(self.get_all_families_zipped())                                    │
│   307 │   │   for i, mobs in enumerate(families):                                                │
│   308 │   │   │   sub_alpha = self.get_sub_alpha(alpha, i, len(families))                        │
│ ❱ 309 │   │   │   self.interpolate_submobject(*mobs, sub_alpha)                                  │
│   310 │                                                                                          │
│   311 │   def interpolate_submobject(                                                            │
│   312 │   │   self,                                                                              │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/animation/creation.py:271 in           │
│ interpolate_submobject                                                                           │
│                                                                                                  │
│   268 │   │   index, subalpha = integer_interpolate(0, 2, alpha)                                 │
│   269 │   │   if index == 0:                                                                     │
│   270 │   │   │   submobject.pointwise_become_partial(outline, 0, subalpha)                      │
│ ❱ 271 │   │   │   submobject.match_style(outline)                                                │
│   272 │   │   else:                                                                              │
│   273 │   │   │   submobject.interpolate(outline, starting_submobject, subalpha)                 │
│   274                                                                                            │
│                                                                                                  │
│ /home/alex/ManimCE/venv/lib/python3.9/site-packages/manim/mobject/types/vectorized_mobject.py:34 │
│ 4 in match_style                                                                                 │
│                                                                                                  │
│    341 │   │   return ret                                                                        │
│    342 │                                                                                         │
│    343 │   def match_style(self, vmobject, family=True):                                         │
│ ❱  344 │   │   self.set_style(**vmobject.get_style(), family=False)                              │
│    345 │   │                                                                                     │
│    346 │   │   if family:                                                                        │
│    347 │   │   │   # Does its best to match up submobject lists, and                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: set_style() got an unexpected keyword argument 'stroke_opacity'

```

</details>

## System specifications

<details><summary>System Details</summary>

- Big Sur M1 and Linux Mint 20.1
- RAM: 8 GB (both)
- Python 8 and 9.
- Installed modules: Only manim's

</details>

**Comment by dawnsqrl** [comment]
This error is very likely due to inconsistent function signature between the `set_style` rewrite of `shape_marchers` and its base function in `VMobject`. The rewrite lacks a considerable number of optional arguments such as the `stroke_opacity` in question, and if added back over the issue is resolved immediately.

I'm not sure why the `shape_marchers`' rewrite prior to this point was leaving out all those arguments, but making only the change above continues to trigger other `AttributeError`s. Some further adjustments may be necessary, or I could eventually figure out that those arguments were omitted for concrete reasons.

**Comment by behackl** [comment]
Fixed via #2658
